src/components/template/TemplateQuestionEditor.tsx

```diff
@@ -87,17 +88,27 @@ export default function TemplateQuestionEditor(props: {
   });
 
   const dependencies = props.data.dependencies;
+  let dependencyComparator =
+    props.data.dependenciesOperator === DependenciesLogicOperator.AND
+      ? '&'
+      : '||';
   const dependencyJsx = dependencies.length ? (
     <>
       <LockIcon className={classes.lockIcon} />
       <ul>
         {dependencies.map((dependency, i) => {
-          const dependencyOperator = i < dependencies.length - 1 ? '&' : '';
+          dependencyComparator =
+            i < dependencies.length - 1 ? dependencyComparator : '';
+
+          const dependenciesAreVisible = !!dependency.dependencyNaturalKey;
 
           return (
-            <li key={dependency.dependencyId + dependency.questionId}>
-              {`${dependency.dependencyNaturalKey} ${dependencyOperator}`}
-            </li>
+            dependenciesAreVisible && (
+              <li key={dependency.dependencyId + dependency.questionId}>
+                {`${dependency.dependencyNaturalKey}`}{' '}
```

Comment:
Could this be rewritten as ``{`${dependency.dependencyNaturalKey} `}``?

Reply:
I guess it could let's see ⏩